sql/clients_daily_v7.sql

```diff
@@ -49,7 +49,44 @@ CREATE TEMP FUNCTION
         key) AS key_value));
   --
 CREATE TEMP FUNCTION
-  udf_aggregate_search_counts(search_counts ARRAY<STRUCT<list ARRAY<STRUCT<element STRUCT<engine STRING,
+  udf_mode_last(list ANY TYPE) AS ((
+    SELECT
+      _value
+    FROM
+      UNNEST(list) AS _value
+    WITH
+    OFFSET
+      AS
+    _offset
+    GROUP BY
+      _value
+    ORDER BY
+      COUNT(_value) DESC,
+      MAX(_offset) DESC
+    LIMIT
+      1 ));
+  --
+CREATE TEMP FUNCTION
+  udf_null_if_empty_list(list ANY TYPE) AS ( IF(ARRAY_LENGTH(list.list) > 0,
```

Comment:
It took me a while to realize that several of these functions expect a struct with a field named "list" rather than expecting an array of elements. I don't want to make these functions more verbose unnecessarily, but perhaps some preliminary commentary about the expected structures in main_summary would be helpful to motivate why these udfs make the assumptions that they do.

Reply:
I'll attempt to add some comments explaining things to the udf files.

if i understand correctly google is turning a parquet schema like 
```
STRUCT<my_field ARRAY<STRING> ... >
```
into
```
STRUCT<my_field STRUCT<list ARRAY<STRUCT<element STRING>> ... >
```
which seems odd at first, but makes some sense when considering the implications an array of arrays not being allowed in BigQuery. For instance `ARRAY_AGG` can't accept array values (there's ARRAY_CONCAT_AGG, which isn't always the right thing, such as with the environment locales lists), and to get nested arrays there has to be a struct at each layer.

So this could all change when we take parquet out of the mix, but it may end up being a useful pattern anyway.